nixos/modules/services/databases/postgresql.nix

```
@@ -287,17 +327,17 @@ in
           ''
             # Create data directory.
             if ! test -e ${cfg.dataDir}/PG_VERSION; then
-              mkdir -m 0700 -p ${cfg.dataDir}
+              mkdir -m ${dirMode} -p ${cfg.dataDir}
               rm -f ${cfg.dataDir}/*.conf
-              chown -R postgres:postgres ${cfg.dataDir}
+              chown -R postgres ${cfg.dataDir}
             fi
           ''; # */
 
         script =
           ''
             # Initialise the database.
             if ! test -e ${cfg.dataDir}/PG_VERSION; then
-              initdb -U ${cfg.superUser}
+              initdb -U ${cfg.superUser} ${lib.concatStringsSep " " cfg.initdbFlags}
```

Combining flags here is more explicit.
Nitpick: `lib.` is also not necessary.

```suggestion
              initdb -U ${cfg.superUser} \
                ${optionalString cfg.groupAccess "--allow-group-access"} \
                ${concatStringsSep " " cfg.initdbFlags}
```

---

I see your point. I'll stick to module-system style though, it's not that bad.

I'll remove `lib.`